docs/proposals/upstream-settings.md

```diff
@@ -0,0 +1,210 @@
+    // +optional
+    Connections *int32 `json"connections,omitempty"`
+
+    // Requests sets the maximum number of requests that can be served through one keep-alive connection.
+    // After the maximum number of requests are made, the connection is closed.
+    // +optional
+    Requests *int32 `json:"requests,omitempty"`
+
+    // Time defines the maximum time during which requests can be processed through one keep-alive connection.
+    // After this time is reached, the connection is closed following the subsequent request processing.
+    // +optional
+    Time *Duration `json:"time,omitempty"`
+
+    // Timeout defines the keep-alive timeout for upstreams.
+    // +optional
+    Timeout *Duration `json:"timeout,omitempty"`
+}
+
+// Duration is a string value representing a duration in time.
+// The format is a subset of the syntax parsed by Golang time.ParseDuration.
+// Examples: 1h, 12m, 30s, 150ms.
+type Duration string
+
+// Size is a string value representing a size. Size can be specified in bytes, kilobytes (suffix k),
+// or megabytes (suffix m).
+// Examples: 1024, 8k, 1m.
+type Size string
+```
+
+Since this Policy only applies to `http` upstreams, there's no way to set the zone size for `stream` upstreams. For now, we can introduce a global `zoneSize` variable in the `NginxProxy` resource that will set the zone size for all upstreams. Then this Policy would override that global setting on upstreams that it attaches to.
+
+### Versioning and Installation
+
+The version of the `UpstreamSettingsPolicy` API will be `v1alpha1`.
+
+The `UpstreamSettingsPolicy` CRD will be installed by the Cluster Operator via Helm or with manifests. It will be required, and if the `UpstreamSettingsPolicy` CRD does not exist in the cluster, NGINX Gateway Fabric will log errors until it is installed.
+
+### Status
+
+#### CRD Label
+
+According to the [Direct Policy Attachment GEP](https://gateway-api.sigs.k8s.io/geps/gep-2648/), the `UpstreamSettingsPolicy` CRD must have the `gateway.networking.k8s.io/policy: direct` label to specify that it is a direct policy.
+This label will help with discoverability and will be used by the planned Gateway API Policy [kubectl plugin](https://gateway-api.sigs.k8s.io/geps/gep-713/#kubectl-plugin-or-command-line-tool).
+
+#### Conditions
+
+#### Conditions/Policy Ancestor Status
+
+According to the [Direct Policy Attachment GEP](https://gateway-api.sigs.k8s.io/geps/gep-2648/), the `UpstreamSettingsPolicy` CRD must include a `status` stanza with a slice of Conditions.
+
+The `Accepted` Condition must be populated on the `UpstreamSettingsPolicy` CRD using the reasons defined in the [PolicyCondition API](https://github.com/kubernetes-sigs/gateway-api/blob/main/apis/v1alpha2/policy_types.go). If these reasons are not sufficient, we can add implementation-specific reasons.
+
+The Condition stanza may need to be namespaced using the `controllerName` if more than one controller could reconcile the Policy.
```

Comment:
Do you think we should make this statement stronger? It's possible to run multiple NGF deployments that both process this policy.